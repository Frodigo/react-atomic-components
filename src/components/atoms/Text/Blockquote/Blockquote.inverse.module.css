.figure {
    composes: figure from './Blockquote.module.css';
    background-color: var(--color-primary);
}

.quote {
    composes: quote from './Blockquote.module.css';
    color: var(--color-white);
}

.caption {
    composes: caption from './Blockquote.module.css';
    color: var(--color-white);
}

.caption::before {
    background-color: var(--color-white);
}

@media (min-width: 768px) {
    .inner {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
        padding: 30px 0;
    }

    .quote {
        font-size: 24px;
        line-height: 32px;
    }
}

@media (min-width: 1300px) {
    .inner {
        max-width: var(--desktop-container-width);
    }
}