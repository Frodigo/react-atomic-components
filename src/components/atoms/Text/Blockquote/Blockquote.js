import React from 'react';
import PropTypes from 'prop-types';

import QuoteIconDefault from '../../../../assets/icons/quote.inline.svg';
import QuoteIconInverse from '../../../../assets/icons/quote.inverse.svg';
import Paragraph from '../Paragraph';

import defaultClasses from './Blockquote.module.css';
import inverseClasses from './Blockquote.inverse.module.css';

const BlockQuote = (props) => {
    const { type, quote, author } = props;

    const classes = type !== 'inverse' ? defaultClasses : inverseClasses;
    const QuoteIcon = type !== 'inverse' ? QuoteIconDefault : QuoteIconInverse;

    return (
        <figure className={classes.figure}>
            <div className={classes.inner}>
                <blockquote className={classes.quote}>
                    <QuoteIcon style={{ marginBottom: '15px' }} />
                    <Paragraph textAlign="center">{quote}</Paragraph>
                </blockquote>
                <figcaption className={classes.caption}>{author}</figcaption>
            </div>
        </figure>
    );
};

BlockQuote.propTypes = {
    author: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['default', 'inverse'])
};

BlockQuote.defaultProps = {
    type: 'default'
};

export default BlockQuote;
