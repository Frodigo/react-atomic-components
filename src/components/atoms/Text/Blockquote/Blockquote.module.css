.figure {
    padding: var(--spacing-xl) var(--spacing-xs);
    text-align: center;
}

.quote {
    color: var(--color-tiertary);
    font-family: var(--font-primary);
    font-size: 18px;
    line-height: 1.5;
    padding: 0 var(--spacing-md);
}

.caption {
    color: var(--color-yellow);
    display: flex;
    flex-direction: column;
    font-size: 18px;
    font-weight: var(--font-weight-bold);
    margin: var(--spacing-xs) 0 0;
}

.caption::before {
    background-color: var(--color-yellow);;
    content: '';
    display: block;
    height: 3px;
    margin: var(--spacing-md) auto;
    width: 20px;
}


@media (min-width: 768px) {
    .inner {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
        padding: 30px 0;
    }

    .quote {
        font-size: 24px;
        line-height: 32px;
    }
}

@media (min-width: 1300px) {
    .inner {
        max-width: var(--desktop-container-width);
    }
}