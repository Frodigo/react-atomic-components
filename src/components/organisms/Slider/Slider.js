import React from 'react';
import PropTypes from 'prop-types';
import { default as SlickSlider } from 'react-slick';

import classes from './Slider.module.css';

const Slider = (props) => {
    const { items } = props;

    const settings = {
        arrows: false,
        dots: true,
        infinite: false,
        speed: 500,
        slidesToShow: 1,
        slidesToScroll: 1,
        autoplay: true,
        autoplaySpeed: 5000
    };

    return (
        <div className={classes.slider}>
            <SlickSlider {...settings}>{items}</SlickSlider>
        </div>
    );
};

Slider.propTypes = {
    items: PropTypes.array.isRequired
};

export default Slider;
