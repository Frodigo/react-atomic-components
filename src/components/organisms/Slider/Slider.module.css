@media (min-width: 1300px) {
    .slider {
       margin-left: -10px;
       margin-right: -10px;
    }
}