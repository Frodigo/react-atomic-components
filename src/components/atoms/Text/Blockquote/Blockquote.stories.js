import React from 'react';

import { default as BlockQuoteComponent } from './Blockquote';

export default {
    title: 'Atoms/Blockquote',
    component: BlockQuoteComponent,
    argTypes: {
        type: {
            control: {
                type: 'select',
                options: ['default', 'inverse']
            }
        }
    }
};

const Template = (args) => <BlockQuoteComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
    quote: `Give a man a program, frustrate him for a day. Teach a man to program, frustrate him for a lifetime.`,
    author: 'Muhammad Waseem'
};

export const Inverse = Template.bind({});
Inverse.args = {
    ...Default.args,
    type: 'inverse'
};
