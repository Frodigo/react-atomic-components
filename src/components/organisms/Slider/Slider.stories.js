import React from 'react';

import Blockquote from '../../atoms/Text/Blockquote';

import { default as SliderComponent } from './Slider';

export default {
    title: 'Organisms/Slider',
    component: SliderComponent
};

const items = [
    <Blockquote key="1" author="Author 1" quote="Lorem ipsum" type="inverse" />,
    <Blockquote key="2" author="Author 2" quote="Lorem ipsum" type="inverse" />,
    <Blockquote key="3" author="Author 3" quote="Lorem ipsum" type="inverse" />
];

const Template = (args) => <SliderComponent {...args} />;

export const Slider = Template.bind({});
Slider.args = {
    items
};
