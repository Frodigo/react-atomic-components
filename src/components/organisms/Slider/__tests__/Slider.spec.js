import React from 'react';
import { render } from '@testing-library/react';

import Slider from '../Slider';
import Blockquote from '../../../atoms/Text/Blockquote';

describe('Slider should be rendered', () => {
    const items = [
        <Blockquote
            key="1"
            author="Author 1"
            quote="Lorem ipsum"
            type="inverse"
        />,
        <Blockquote
            key="2"
            author="Author 2"
            quote="Lorem ipsum"
            type="inverse"
        />,
        <Blockquote
            key="3"
            author="Author 3"
            quote="Lorem ipsum"
            type="inverse"
        />
    ];

    test('Image carousel renders correctly', () => {
        const { container } = render(<Slider items={items} />);

        expect(container).toMatchSnapshot();
    });
});
