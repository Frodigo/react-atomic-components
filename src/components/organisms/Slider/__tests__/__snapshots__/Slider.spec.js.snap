// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should be rendered Image carousel renders correctly 1`] = `
<div>
  <div
    class="slider"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <figure
                class="figure"
              >
                <div
                  class="inner"
                >
                  <blockquote
                    class="quote"
                  >
                    <span>
                      svg
                    </span>
                    <p
                      class="paraghaph"
                      style="text-align: center;"
                    >
                      Lorem ipsum
                    </p>
                  </blockquote>
                  <figcaption
                    class="caption"
                  >
                    Author 1
                  </figcaption>
                </div>
              </figure>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <figure
                class="figure"
              >
                <div
                  class="inner"
                >
                  <blockquote
                    class="quote"
                  >
                    <span>
                      svg
                    </span>
                    <p
                      class="paraghaph"
                      style="text-align: center;"
                    >
                      Lorem ipsum
                    </p>
                  </blockquote>
                  <figcaption
                    class="caption"
                  >
                    Author 2
                  </figcaption>
                </div>
              </figure>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <figure
                class="figure"
              >
                <div
                  class="inner"
                >
                  <blockquote
                    class="quote"
                  >
                    <span>
                      svg
                    </span>
                    <p
                      class="paraghaph"
                      style="text-align: center;"
                    >
                      Lorem ipsum
                    </p>
                  </blockquote>
                  <figcaption
                    class="caption"
                  >
                    Author 3
                  </figcaption>
                </div>
              </figure>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
